### PR TITLE
Replace checkPaymentChannelBalance panic with error

### PR DESCRIPTION
**Pull request by bitwiseguy**
resolves https://github.com/statechannels/boost/issues/9

### CODE REVIEW

File: cmd/booster-http/server.go [hunk 1]
```diff
@@ -203,8 +203,14 @@ func (s *HttpServer) handleByPieceCid(w http.ResponseWriter, r *http.Request) {
...
-		if hasPaid := checkPaymentChannelBalance(s.nitroRpcClient, chId, expectedPaymentAmount); !hasPaid {
-			writeError(w, r, http.StatusPaymentRequired, "payment required")
+		hasPaid, err := checkPaymentChannelBalance(s.nitroRpcClient, chId, expectedPaymentAmount)
+		if err != nil {
+			writeError(w, r, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		if !hasPaid {
+			webError(w, fmt.Errorf("payment of %d required", expectedPaymentAmount.Uint64()), http.StatusPaymentRequired)
```
Comment (geoknee) [inline cmd/booster-http/server.go:213]: What's the difference between `writeError` and `webError`?

**Review by geoknee — APPROVED** [review]
👍

File: cmd/booster-http/server.go (hunk 1 above)
Comment (bitwiseguy) [reply cmd/booster-http/server.go:213]: It looks like `webError` sets some additional headers related to CORS. I am changing this `webError` to `writeError` to conform to the rest of the error handling within this method.

**Comment by geoknee** [comment]
@bitwiseguy the base branch was rebased, which is giving you the conflicts. Seeing as you are out and that's my fault, I'm happy to sort it out?

**Comment by geoknee** [comment]
> @bitwiseguy the base branch was rebased, which is giving you the conflicts. Seeing as you are out and that's my fault, I'm happy to sort it out?

#12

**Pull request closed without merging**